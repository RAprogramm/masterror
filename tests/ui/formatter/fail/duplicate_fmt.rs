// SPDX-FileCopyrightText: 2025 RAprogramm <andrey.rozanov.vl@gmail.com>
//
// SPDX-License-Identifier: MIT

use masterror::Error;

#[derive(Debug, Error)]
#[error(fmt = crate::format_error, fmt = crate::format_error)]
struct DuplicateFmt;

fn format_error(
    _error: &DuplicateFmt,
    f: &mut core::fmt::Formatter<'_>
) -> core::fmt::Result {
    f.write_str("duplicate")
}

fn main() {}
