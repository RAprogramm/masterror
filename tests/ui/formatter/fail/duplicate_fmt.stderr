error: duplicate `fmt` handler specified
 --> tests/ui/formatter/fail/duplicate_fmt.rs:4:42
  |
4 | #[error(fmt = crate::format_error, fmt = crate::format_error)]
  |                                          ^^^^^
